components/twitter/twitter.app.js

```diff
@@ -818,19 +885,212 @@ module.exports = {
+        }
+      })).data
+    },
+    async lookupUsers(userIdArray) {   
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/users/lookup.json`,
+        params: {
+          user_id: userIdArray.join(),
+        }
+      })).data
+    },
+    async search(q, since_id, tweet_mode, count, result_type, lang, locale, geocode, max_id) {   
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/search/tweets.json`,
+        params: {
+          q,
+          since_id,
+          max_id,
+          tweet_mode,
+          count,
+          result_type,
+          lang,
+          locale,
+          geocode,
+        }
+      }))
+    },
+    async getTrendLocations() {   
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/trends/available.json`,
+      })).data
+    },
+    async getTrends(opts = {}) {   
+      const {
+        id = 1,
+      } = opts
+      return (await this._makeRequest({
+        url: `https://api.twitter.com/1.1/trends/place.json`,
+        params: {
+          id,
+        }
+      })).data
+    },
+    async searchHelper(opts = {}) {
+      const tweets = []
+
+      const {
+        tweet_mode = 'extended',
         result_type,
+        count = 100,
         lang,
         locale,
         geocode,
-      })
+        enrichTweets = true,
+        includeReplies = true,
+        includeRetweets = true,
+      } = opts
+
+      let { q, max_id, since_id = 0 } = opts
+      let min_id
+
+      if(includeReplies === false) {
+        q = `${q} -filter:replies`
+      }
+  
+      if(includeRetweets === false) {
+        q = `${q} -filter:nativeretweets`
+      }
+  
+      const response = await this.search(q, since_id, tweet_mode, count, result_type, lang, locale, geocode, max_id)
+      
+      //console.log(response)
+      if (_.get(response, 'status', 'Error') === 200) {
+        if (!max_id) {
+          max_id = since_id || 0
+        }
+    
+        for (let tweet of response.data.statuses) {
+          if (tweet.id_str !== since_id && tweet.id_str !== max_id) {
+            if (enrichTweets) {
+              tweet.created_at_timestamp = moment(tweet.created_at, 'ddd MMM DD HH:mm:ss Z YYYY').valueOf()
+              tweet.created_at_iso8601 = moment(tweet.created_at, 'ddd MMM DD HH:mm:ss Z YYYY').toISOString()
+            }
+            tweets.push(tweet)
+            if (tweet.id_str > max_id) {
+              max_id = tweet.id_str
+              if(!min_id) {
+                min_id = max_id
+              }
+            }
+            if (tweet.id_str < max_id) {
+              min_id = tweet.id_str
+            }
+          }
+        }
+
+        return {
+          tweets,
+          max_id,
+          min_id,
+          count,
+          resultCount: response.data.statuses.length,
+          statusCode: response.status, 
+        }
+      } else {
+        console.log(`Last request was not successful.`)
```

Comment:
See comment above about status check... should just more explicitely do

```
const resp = await this.search(...)
if (!resp) {
  // this code block
  return { statusCode: "Error" } // or whatever
}
// handle the stuff above but not nested in else
```

Reply:
updated. need to test.